synapse/federation/federation_base.py

```diff
@@ -55,13 +51,15 @@ def __init__(self, hs):
         self.store = hs.get_datastore()
         self._clock = hs.get_clock()
 
-    def _check_sigs_and_hash(self, room_version: str, pdu: EventBase) -> Deferred:
+    def _check_sigs_and_hash(
```

Comment:
Not saying that we should undo it (it's certainly much cleaner now), but for the record: this seems to go beyond what is needed to fix the issue here. afaict only the calls to `_check_sigs_and_hash_and_fetch` needed changing; `_check_sigs_and_hash`, `_check_sigs_and_hashes` and `_check_sigs_on_pdus` were ok.

Reply:
Yes, it does go a bit beyond the minimum, but it seemed clearer / more efficient to not switch back and forth between `RoomVersion` and a `str` repeatedly.

Reply:
yup, fair.